src/runtime_src/core/edge/common_em/config.cxx

```diff
@@ -884,4 +885,46 @@ namespace xclemulation{
     ifs.close();
     return false;
   }
+
+  int getIPName2Index(const char *name,const char* buffer)
+  {
+    std::string errmsg;
+    const uint64_t bad_addr = 0xffffffffffffffff;
```

Comment:
yes, modified code as suggested

Reply:
It has always hurt my eyes to see assignment of -1 to an unsigned integral.   Better to use `std::numeric_limits<uint64_t>::max(); `  Don't bother changing, just a comment

Reply:
You are right. Let's say `~0U` as a middle ground?
It is short and unsigned. :-)
Otherwise, while `-1` looks strange, this is correct with unsigned because the semantics is well defined.
Good news for hardware people: C++20 defines the representation of signed numbers as 2-complement, so it legalizes a lot of usual atrocities... :-)

Reply:
Uh no.  Exactly my worry.  ` ~0U != std::numeric_limits<uint64_t>::max()`  Please be specific.
Agreed with -1, but actually I think VC++ barfs on that.

Reply:
Yes, you are right. I introduce a bug here for not using a double `UU`. :-(
Your suggestion with `auto constexpr bad_addr = std::numeric_limits<uint64_t>::max();` seems the right answer!